ogre2/src/Ogre2BoundingBoxCamera.cc

```diff
@@ -0,0 +1,1527 @@
+    /*
+
+        1 -------- 0
+       /|         /|
+      2 -------- 3 .
+      | |        | |
+      . 5 -------- 4
+      |/         |/
+      6 -------- 7
+
+    */
+
+    std::vector<math::Vector2d> clippedVertices;
+    const math::Vector4d box(-1.0, -1.0, 1.0, 1.0);
+    // top
+    this->dataPtr->AddToViewportLines(box, vertices2d[0], vertices2d[1],
+        clippedVertices);
+    this->dataPtr->AddToViewportLines(box, vertices2d[1], vertices2d[2],
+        clippedVertices);
+    this->dataPtr->AddToViewportLines(box, vertices2d[2], vertices2d[3],
+        clippedVertices);
+    this->dataPtr->AddToViewportLines(box, vertices2d[3], vertices2d[0],
+        clippedVertices);
+    // bottom
+    this->dataPtr->AddToViewportLines(box, vertices2d[4], vertices2d[5],
+        clippedVertices);
+    this->dataPtr->AddToViewportLines(box, vertices2d[5], vertices2d[6],
+        clippedVertices);
+    this->dataPtr->AddToViewportLines(box, vertices2d[6], vertices2d[7],
+        clippedVertices);
+    this->dataPtr->AddToViewportLines(box, vertices2d[7], vertices2d[4],
+        clippedVertices);
+    // pillars
+    this->dataPtr->AddToViewportLines(box, vertices2d[0], vertices2d[4],
+        clippedVertices);
+    this->dataPtr->AddToViewportLines(box, vertices2d[1], vertices2d[5],
+        clippedVertices);
+    this->dataPtr->AddToViewportLines(box, vertices2d[2], vertices2d[6],
+        clippedVertices);
+    this->dataPtr->AddToViewportLines(box, vertices2d[3], vertices2d[7],
+        clippedVertices);
+
+    // Convert To screen coord.
+    uint32_t width = this->ImageWidth();
+    uint32_t height = this->ImageHeight();
+
+    std::vector<math::Vector2i> projVertices;
+    for (auto &vertex : clippedVertices)
+    {
+      // convert from [-1, 1] range to [0, 1] range & to the screen range
+      vertex.X() = uint32_t((vertex.X() + 1.0) / 2 * width );
+      vertex.Y() = uint32_t((1.0 - vertex.Y()) / 2 * height);
+
+      vertex.X() = std::max<uint32_t>(0, vertex.X());
+      vertex.Y() = std::max<uint32_t>(0, vertex.Y());
+      vertex.X() = std::min<uint32_t>(vertex.X(), width - 1);
+      vertex.Y() = std::min<uint32_t>(vertex.Y(), height - 1);
+
+      projVertices.push_back(math::Vector2i(vertex.X(), vertex.Y()));
+    }
+
+    for (unsigned int endPt = 0; endPt < projVertices.size(); endPt += 2)
+    {
+      this->DrawLine(_data, projVertices[endPt], projVertices[endPt + 1],
+          _color);
+    }
+
+    return;
+  }
+
+  // 2D box
+  math::Vector2 minVertex(_box.Center().X() - _box.Size().X() / 2,
+    _box.Center().Y() - _box.Size().Y() / 2);
+
+  math::Vector2 maxVertex(_box.Center().X() + _box.Size().X() / 2,
+    _box.Center().Y() + _box.Size().Y() / 2);
+
+  uint32_t width = this->ImageWidth();
+
+  std::vector<uint32_t> x_values =
+    {uint32_t(minVertex.X()), uint32_t(maxVertex.X())};
+  std::vector<uint32_t> y_values =
+    {uint32_t(minVertex.Y()), uint32_t(maxVertex.Y())};
```

Comment:
Looks like the issue has to do with calculating the maxVertex. 

The example crashes when the bottom boundary line touches the bottom of the screen. Unlike the top, when the line reaches the bottom of the screen it either disappears or crashes the application.

Setting this line to the following code fixes the crash and also bounds the bottom line to the screen.
```
    {uint32_t(minVertex.Y()), uint32_t(maxVertexY() - 1)};
```